### PR TITLE
Added 2 new AWS Odin Detections

### PR DESCRIPTION
**Pull request by hbenac10**
Background
MITRE T1525 CloudTrail Detections

[MITRE IaaS Gap] - T1578 - Modify Cloud Compute Infrastructure
[MITRE IaaS Gap] - T1498 - regular detection - WAF disassociate

Changes
New detections

### CODE REVIEW

File: rules/aws_cloudtrail_rules/aws_modify_cloud_compute_infrastructure.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
+AnalysisType: rule
+Description: Detection when the cloud compute infrastructure is modified.
+DisplayName: AWS Modify Cloud Compute Infrastructure
+Enabled: true
+Filename: aws_modify_cloud_compute_infrastructure.py
+Reference: https://attack.mitre.org/techniques/T1578/
+Severity: High
+Tags:
+    - 'MITRE ATT&CK: T1578'
```
Comment (darwayne) [inline rules/aws_cloudtrail_rules/aws_modify_cloud_compute_infrastructure.yml:9]: You'll want this to match [this format](https://github.com/panther-labs/panther-analysis/blob/cd220c87982011d4ad156c7daecd2857c358d154/rules/osquery_rules/osquery_linux_logins_non_office.yml#L13-L15) for this to show in the MITRE Matrix

```suggestion
Reports:
  MITRE ATT&CK:
    - TA0005:T1578
```

File: rules/aws_cloudtrail_rules/aws_waf_disassociation.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
+AnalysisType: rule
+Description: Detection to alert when a WAF disassociates from a source.
+DisplayName: AWS WAF Disassociation
+Enabled: true
+Filename: aws_waf_disassociation.py
+Reference: https://attack.mitre.org/techniques/T1078/
+Severity: Critical
+Tags:
+    - 'MITRE ATT&CK: TA0004:T1078'
```
Comment (darwayne) [inline rules/aws_cloudtrail_rules/aws_waf_disassociation.yml:9]: You'll want this to match [this format](https://github.com/panther-labs/panther-analysis/blob/cd220c87982011d4ad156c7daecd2857c358d154/rules/osquery_rules/osquery_linux_logins_non_office.yml#L13-L15) for this to show in the MITRE Matrix

```suggestion
Reports:
  MITRE ATT&CK:
    - TA0004:T1078
```

**Review by edyesed — CHANGES_REQUESTED** [review]
IMO, we should separate out the aws_modify_cloud_compute detection from this PR. As currently written, the modify_compute detection will fire for every ec2:TerminateInstance command that happens in the cloudtrail, which will lead to alert fatigue in ec2 rich environments.

**Comment by edyesed** [comment]
replaced by #532

**Pull request closed without merging**